with

```diff
@@ -20,20 +20,20 @@ while [ true ]; do
   while IFS="" read -r -e -d $'\n' -p "$pmpt " options; do
     history -s "$options"
     
-    curr_command="$(echo $options | { read first rest ; echo $first ; })" 
+    curr_command="$({ read first rest ; echo $first ; } <<<"$options")" 
```

Comment:
In the spirit of Occam's razor, that they're not necessary. You can say any of

    process <<<a
    echo a |process
    echo a |cat |process
    echo a |cat |tac |cat |process

or any of the other infinite possible ways out there to get the same task done. How do you choose which way to go about this?

What's wrong echo? Nothing, you can use echo. What's wrong with cat? Nothing, you can use that. Or tac, wonderful utility, you can use that too. There's nothing wrong with any of them. They're just not necessary here, any of them. And that's what's wrong with this. Doing something that's not necessary. Writing more code than you must.

Okay, I've clarified my point I think. If this is an opinionated matter (can I say in my opinion it isn't?) then you can close this.